lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -453,9 +453,25 @@ sub show ($self) {
     my $job_id = int($self->stash('jobid'));
     my $details = $self->stash('details') || 0;
     my $check_assets = !!$self->param('check_assets');
-    my $job = $self->schema->resultset('Jobs')->find($job_id, {prefetch => 'settings'});
-    return $self->reply->not_found unless $job;
+    my $follow = $self->param('follow');    # follow job clones and report most recent result for given id
+
+    my @cloned_as;
+    my $job;
+    while (1) {
+        return unless $job = $self->find_job_or_render_not_found($job_id, {prefetch => 'settings'});
+        if ($follow && $job->clone_id) {
+            push(@cloned_as, $job->clone_id);
+            $job_id = $job->clone_id;
+            next;
```

Comment:
This should be its own function, like the `latest_job` function we already have (but tracking the whole chain of IDs). Especially since you're duplicating it later on. We could also simplify this to use the existing `latest_job` function and not return the whole chain of IDs.

You could also use *one* recursive SQL query instead of doing one per clone. Search for `sub descendants` or `sub incomplete_ancestors` for functions where we already do this. This would just be an optimization but maybe a worthwhile one and the SQL code for this is actually quite interesting.